### PR TITLE
Remove traces of 0-limb support from bignum core

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
The bignum core module [does not handle limb arrays of size 0](https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-3.4.0/library/bignum_core.h#L36), to keep it small and simple. There are a few functions that still have explicit support for 0 limbs, because they date from the very early days before we settle on the semantics. Arguably these checks are defensive, however in bignum core we are deliberately not very defensive because prototyping showed us that it significantly hurt code size. So remove the explicit handling for 0-limb arrays (where it was easy to find). This makes the code a little easier to read (and very very slightly smaller — by 6 Thumb instructions in our basic reference build).

## Gatekeeper checklist

- [x] **changelog** no (internal change)
- [x] **backport** no (library enhancement)
- [x] **tests** no (refactoring only)

### CODE REVIEW

File: library/bignum_core.c [hunk 1]
```diff
@@ -53,10 +53,6 @@ size_t mbedtls_mpi_core_bitlen(const mbedtls_mpi_uint *A, size_t A_limbs)
 {
     size_t i, j;
 
-    if (A_limbs == 0) {
```
Comment (daverodgman) [inline library/bignum_core.c:56]: This part conflicts with #7462

**Review by daverodgman — CHANGES_REQUESTED** [review]

File: library/bignum_core.c (hunk 1 above)
Comment (gilles-peskine-arm) [reply library/bignum_core.c:56]: No problem, I'll rebase once 7462 is merged

Comment (gilles-peskine-arm) [reply library/bignum_core.c:56]: CI says one of the checks I removed was actually necessary, anyway...

**Comment by tom-daubney-arm** [comment]
We are now converting older PRs to draft PRs where the following conditions are met: They have not been updated in the last 3 months, and they need work to complete.

**Comment by tom-cosgrove-arm** [comment]
@gilles-peskine-arm Do we actually want this PR at all? ISTR it wasn't possible to remove all of the traces of 0-limb support safely anyway; it feels like there should have been a statement at the beginning about this, and it included during development of `bignum_core.c` if we really wanted to do this

**Pull request closed without merging**